src/components/PostDetails.tsx

```diff
@@ -1,117 +1,253 @@
-import React from 'react';
+import React, {
+  useState, useEffect, ChangeEvent, useCallback, FormEvent,
+} from 'react';
+import classNames from 'classnames';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { SelectedPost } from '../types/SelectedPost';
+import { Comment, CommentData } from '../types/Comment';
+import { createComment, getComments, deleteComment } from '../api/comments';
+import { FieldErrors } from '../types/Errors';
+
+interface Props {
+  selectedPost: SelectedPost | null
+}
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [newComment, setNewComment] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+  const [fieldErrors, setFieldError] = useState<FieldErrors>({
+    name: false,
+    email: false,
+    body: false,
+  });
+
+  const [error, setError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isWriting, setIsWriting] = useState<boolean>(false);
+
+  const clearFormField = useCallback(() => {
+    setNewComment({
+      name: '',
+      email: '',
+      body: '',
+    });
+
+    setFieldError({
+      name: false,
+      email: false,
+      body: false,
+    });
+  }, [comments]);
+
+  const newCommentId = useCallback(() => {
+    const commentIds = comments.map(comment => comment.id);
+
+    return (Math.max(...commentIds) + 1);
+  }, [comments]);
+
+  const isEmailValid = useCallback((email: string) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    return emailRegex.test(email.trim());
+  }, [newComment]);
```

Comment:
The useCallback dependency array should be empty since 'isEmailValid' does not depend on the 'newComment' state.